### PR TITLE
Add resetState fucntionality

### LINKED ISSUE #162: Add ability to reset state without reinitializing

### PR DESCRIPTION
**Pull request by TerrenceLJones**
Also update documentation (and remove references to `StateClass` references). This resolves #162.

### CODE REVIEW

**Comment by TerrenceLJones** [comment]
Hey @paulfalgout can you pay special attention to the doc? Some of the wording might need to be adjusted.

File: src/mixins/state.js [hunk 1]
```diff
@@ -110,6 +110,21 @@ export default {
...
+   *
+   * @public
+   * @method resetState
+   * @returns {Backbone.Model|*} - The _stateModel or the attribute value of the _stateModel
+   */
+
+  resetState() {
+    const defaults = this._stateModel.defaults;
+
+    return this._stateModel.set.apply(this._stateModel, defaults);
```
Comment (paulfalgout) [inline src/mixins/state.js]: I know we used the apply somewhere.. but I think in this case it isn't apply.. it's just `this._stateModel.set(defaults)`?  but defaults can be a function.. so it needs to be `const defaults = _.result(this._stateModel, 'defaults');`

Definitely needs a test or two.. maybe for when defaults is a hash or a function.. maybe one if it doesn't exist.

File: docs/mixins/state.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Use a `StateMixin` if your object/view needs to maintain information that isn't
 
 * [Using StateMixin](#using-statemixin)
 * [StateMixin's `StateModel`](#statemixins-statemodel)
-* [StateMixin's `stateDefaults`](#statemixins-statedefaults)
```
Comment (paulfalgout) [inline docs/mixins/state.md]: Maybe move this item up about `StateModel`?  Both `StateModel` and `stateEvents` are options.  This item should probably just say `[Setting a default state]` vs `StateMixin's` as this isn't specific to the StateMixin

File: docs/mixins/state.md [hunk 3]
```diff
@@ -154,7 +155,7 @@ implementations of `modelEvents` and `collectionEvents` in the [Marionette.View]
 ### Setting State
 
 `StateMixin` has a `setState` method that exposes the `Backbone.Model.set`
-for the `StateClass`'s attached `StateModel`.  Implementation will match [Backbone.Model.set](http://backbonejs.org/#Model-set) documentation.
+for the `StateMixin`'s attached `StateModel`.  Implementation will match [Backbone.Model.set](http://backbonejs.org/#Model-set) documentation.
```
Comment (paulfalgout) [inline docs/mixins/state.md]: Is `StateClass` anywhere else?

File: docs/mixins/state.md [hunk 4]
```diff
@@ -170,10 +171,33 @@ var MyToolKitApp = new Marionette.Toolkit.App({
 myToolKitApp.setState('foo', 'bar');
 ```
 
+### Resetting State
+
+`StateMixin` has a `resetState` method that sets the `StateModel` instance attributes back to the defined defaults.  Implementation will match [Backbone.Model.defaults](http://backbonejs.org/#Model-defaults) documentation.
```
Comment (paulfalgout) [inline docs/mixins/state.md]: maybe link `[defined defaults]` back up to the setting default info

File: src/mixins/state.js (hunk 1 above)
Comment (TerrenceLJones) [inline src/mixins/state.js]: Yea that makes sense on both parts.

File: docs/mixins/state.md (hunk 3 above)
Comment (TerrenceLJones) [inline docs/mixins/state.md]: Yes...and in my search I found that the core `readme.md` had not been updated. 😞

File: README.md [hunk 5]
```diff
@@ -12,15 +12,15 @@ for complex large scale application problems.  This toolkit was
...
 
 ## Documentation
-  [StateClass](./docs/state-class.md) - `Marionette.Object` with a `Backbone.Model` for keeping state.
+  [App](./docs/app.md) - An extension of `Marionette.Object`. Its purpose is to provide an object with a `initialize`/`start`/`stop`/`destroy` lifecycle.
 
-  [Component](./docs/component.md) - Reusable `StateClass` with `View` management boilerplate
+  [Component](./docs/component.md) - Provides a consistent interface for which to package state-view-logic.
 
-  [AbstractApp](./docs/abstract-app.md) - Object including a start/stop lifecycle and running event registry.
+  [StateMixin](./docs/mixings/state.md) - `Marionette.Object` with a `Backbone.Model` for keeping state.
```
Comment (paulfalgout) [inline README.md]: This isn't a Marionette.Object though.. it's just an object for mixing into other things.

Comment (paulfalgout) [inline README.md]: also "mixings"

File: README.md [hunk 6]
```diff
@@ -12,15 +12,15 @@ for complex large scale application problems.  This toolkit was
...
 
-  [App](./docs/app.md) - `AbstractApp` with mixed in "App Manager" for adding and removing child `App`s
+  [ChildAppsMixin](./docs/mixins/child-apps.md) - Functionality to add or remove child Apps to a parent App and connects the child App lifecycle with the parent App lifecycle.
 
-  [Asnyc App start](./docs/async-app-start.md) - How to easily start apps asynchronously
```
Comment (paulfalgout) [inline README.md]: We should definitely keep the async app info.

**Pull request merged**